sprite.h

```diff
@@ -29,20 +29,41 @@ class Sprite : public Identifiable<Empty> {
 	}
 
 	Point &home();
+	inline static bool allow_screen_wrapping = true;
```

Comment:
No. No behavior (ESPECIALLY control flow) dictated by global variables. A static variable should pretty much always be constant. 

Now that we have to decide whether or not the screen wrapping takes place on a pattern-by-pattern basis, this means the responsibility for wrapping or clamping the sprite has now fallen to the sprite coordinator, NOT the sprite itself. I think it goes somewhere after `m_current_player->update()` in `SpriteChoreographer::update()`.

Reply:
I'll just quickly mention that my first idea was for it to check the current pattern, but couldn't figure out how to make it work without changing approximately 50 different things.

However, I was also not changing the location of the wrap function.